test/optim/test_zeroptimizer.py

```diff
@@ -0,0 +1,469 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All rights reserved.
```

Comment:
We will need to add the test files to [run_test.py](https://github.com/pytorch/pytorch/blob/83d358da7c9957db6cc6456f69f43a4c2a4d54e4/test/run_test.py) to enable them in CI. Also, let's move this test to the distributed folder. Please refer to `https://github.com/pytorch/pytorch/blob/master/test/distributed/test_c10d.py` as an example.